czkawka_core/src/common_dir_traversal.rs

```diff
@@ -546,9 +548,17 @@ fn process_symlink_in_symlink_mode(
     fe_result.push(fe);
 }
 
-pub fn common_read_dir(current_folder: &Path, warnings: &mut Vec<String>) -> Option<ReadDir> {
+pub fn common_read_dir(current_folder: &Path, warnings: &mut Vec<String>) -> Option<Vec<Result<DirEntry, std::io::Error>>> {
     match fs::read_dir(current_folder) {
-        Ok(t) => Some(t),
+        Ok(t) => {
+            // Make directory traversal order stable
```

Comment:
Not sure if this will not have big performance impact with bigger folders, because comparing big number of paths are quite time consuming(paths are compared by each component alone)